rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -912,11 +944,72 @@ void main_window::DecryptSPRXLibraries()
 				{
 					gui_log.error("Failed to create %s", new_path);
 				}
+
+				break;
 			}
-			else
+			else if (!invalid)
 			{
-				gui_log.error("Failed to decrypt %s", old_path);
+				// Allow the user to manually type KLIC if decryption failed
+
+				if (tried)
+				{
+					gui_log.error("Failed to decrypt %s with specfied KLIC, retrying", old_path);
+				}
+
+				const std::string filename = old_path.substr(old_path.find_last_of(fs::delim) + 1);
+
+				input_dialog dlg(32, QString{}, tr("Enter KLIC of %0").arg(qstr(filename)),
+					tr("Hexadecimal only."), "00000000000000000000000000000000", this);
+
+				QFont mono = QFontDatabase::systemFont(QFontDatabase::FixedFont);
+				mono.setPointSize(8);
+				dlg.set_input_font(mono, true, '0');
+				dlg.set_clear_button_state(false);
+				dlg.set_validator(new QRegExpValidator(QRegExp("^[a-fA-F0-9]*$"))); // HEX only
+
+				if (dlg.exec() == QDialog::Accepted)
+				{
+					const std::string text = sstr(dlg.get_input_text());
+
+					if (text.size() == 32)
+					{
+						auto& klic = (tried ? klics.back() : klics.emplace_back());
+
+						// It must succeed (only hex characters are present)
+						std::from_chars(&text[0], &text[16], klic._u64[1], 16); // Not a typo: on LE systems the u64[1] part will be swapped with u64[0] later
+						std::from_chars(&text[16], &text[32], klic._u64[0], 16); // And on BE systems it will be already swapped by index internally
+
+						// Needs to be in big endian because the left to right byte-order means big endian
+						klic = std::bit_cast<be_t<v128>>(klic);
+
+						// Retry with specified KLIC
+						key_it -= +std::exchange(tried, true); // Rewind on second and above attempt
+						gui_log.notice("KLIC entered for %s: %s", filename, klic);
+						continue;
+					}
+					else
+					{
+						if (!tried)
+						{
+							tried = true;
+							klics.emplace_back(); // On further attempts this value will be edited
+							key_it++; // Skip it
+						}
+
+						gui_log.error("KLIC is invalid. (%s)"
+							"\nHint: KLIC (KLicense key) is a 16-byte long string. (32 hexadecimal characters)"
```

Comment:
now that i think about it.... It's better if you use this Hint as "text" arg in the input_dialog.
It doesn't make much sense in the log...

Reply:
Ok added as label text on second and more try.